include/crm/common/xml_internal.h

```diff
@@ -213,6 +213,21 @@ const char *pcmk__xe_add_last_written(xmlNode *xe);
 xmlNode *pcmk__xe_match(const xmlNode *parent, const char *node_name,
                         const char *attr_n, const char *attr_v);
 
+/*!
+ * \internal
+ * \brief Find first XML child element with given name
+ *
+ * \param[in] parent  XML element to search
+ * \param[in] name    Name that child must match (\c NULL for any)
+ *
+ * \return Matching XML child element, or \c NULL if none found
+ */
+static inline xmlNode *
+pcmk__xe_match_name(const xmlNode *parent, const char *name)
+{
+    return pcmk__xe_match(parent, name, NULL, NULL);
+}
```

Comment:
I'm not a fan of the `match` naming (even though I came up with it) since it doesn't indicate that we're matching a child, not the given element. Maybe `pcmk__xe_find_child()`. I'm not sure the wrapper adds much value, we can just use the one function everywhere, although it would make for loops uglier.

Reply:
`pcmk__xe_first_child(...)` might be more readable, especially in loop context.

Reply:
`pcmk__xe_first_child()` already exists -- it's a convenience function that doesn't take any match criteria. It behaves the same as `pcmk__xe_match(parent, NULL, NULL, NULL)`.

I think convenience functions like the current `pcmk__xe_first_child()` and `first_named_child()`/`pcmk__xe_match_name()` are useful, especially in loops. Naming is pretty tricky without getting too long though.

Reply:
I like `pcmk__xe_first_child()` for the one function to rule them all

Reply:
If there are non-NULL arguments it's pretty obvious that it's the first child of a certain type

Reply:
> If there are non-NULL arguments it's pretty obvious that it's the first child of a certain type

Definitely. Having only `pcmk__xe_first_child()` makes sense, it's just inconvenient for the common cases where we have no match criteria or only name as the match criterion. Not a deal-breaker (especially due to the naming difficulties mentioned above).

Reply:
I'm working on this, and it's turning out to be pretty damn clunky. If we do decide to keep convenience wrappers, I propose:
* `pcmk__xe_first_child_any(parent)`
* `pcmk__xe_first_child_named(parent, name)` or `pcmk__xe_first_child_name(parent, name)`